### PR TITLE
Requeriment: Token auth

### PR DESCRIPTION
**Pull request by lucasVoboril**
# Before
The project was configured with rubocop, travis, devise

# Requeriment
Allow a user to log in with devise_token_auth.

# Changes
The gem 'devise_token_auth' was incorporated, and apply changes

# After
The project is configured with rubocop, travis, devise, and allows the authentication of a user with devise_token_auth

# Autor and date
Lucas Voboril, 4 may 2018

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -43,6 +43,8 @@ gem 'rubocop-rspec', '1.10.0'
 # Devise to authenticate users
 gem 'devise', '~> 4.4.3'
 
+gem 'devise_token_auth'
```
Comment (Martum) [inline Gemfile]: Remember to always lock the version!

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,10 @@
 Rails.application.routes.draw do
-  devise_for :user
+  mount_devise_token_auth_for 'User', at: 'auth'
+  resources :users
```
Comment (Martum) [inline config/routes.rb]: We don't need any of these routes definitions! Lets keep only the one which mounts devise token auth

**Review by Martum — CHANGES_REQUESTED** [review]

File: config/routes.rb [hunk 3]
```diff
@@ -1,3 +1,9 @@
 Rails.application.routes.draw do
-  devise_for :user
+  mount_devise_token_auth_for 'User', at: 'auth'
+  
+  get 'home/index'
+  get 'user/index'
+  
+  root to: "home#index"
```
Comment (Martum) [inline config/routes.rb]: With my previous comment I meant we don't need any of these routes
```
get 'home/index'
get 'user/index'
root to: "home#index"
```
Lets just keep the mount of devise token auth!

Comment (lucasVoboril) [reply config/routes.rb]: My bad ... What happened is that github only painted me two lines

**Review by Martum — APPROVED** [review]
Merge it and rebase all the branches! Yes, ALL the branches!!!

**Pull request merged**